src/app/demo/demo/demo.component.ts

```diff
@@ -21,17 +22,23 @@ export class DemoComponent {
 
   date: Moment;
   dates: Moment[] = [];
-
-  formGroup: FormGroup = new FormGroup({
-    datePicker: new FormControl(this.date),
-  });
   material: boolean = true;
   required: boolean = false;
   disabled: boolean = false;
   validationMinDate: Moment;
   validationMaxDate: Moment;
   placeholder: string = 'Choose a date...';
 
+  formGroup: FormGroup = new FormGroup({
+    datePicker: new FormControl(this.date, [
+      this.required ? Validators.required : () => undefined,
+      control => this.validationMinDate && this.config && moment(control.value, this.config.format).isBefore(this.validationMinDate)
```

Comment:
I suppose you could have the library export the validation functions.  That would make it easier on users of the library.